pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+
+	jsonpatch "gomodules.xyz/jsonpatch/v2"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// WithLosslessDefaulter creates a new Handler for a CustomDefaulter interface that **drops** remove operations,
+// which are typically the result of new API fields not present in Kueue libraries.
+func WithLosslessDefaulter(scheme *runtime.Scheme, obj runtime.Object, defaulter admission.CustomDefaulter) admission.Handler {
+	return &losslessDefaulter{
+		Handler: admission.WithCustomDefaulter(scheme, obj, defaulter).Handler,
+	}
+}
+
+type losslessDefaulter struct {
+	admission.Handler
+}
+
+// Handle handles admission requests, **dropping** remove operations from patches produced by controller-runtime.
+// The controller-runtime handler works by creating a jsondiff from the raw object and the marshalled
+// version of the object modified by the defaulter. This generates "remove" operations for fields
+// that are not present in the go types, which can occur when Kueue libraries are behind the latest
+// released CRDs.
+// Dropping the "remove" operations is safe because Kueue's job mutators never remove fields.
+func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) admission.Response {
+	response := h.Handler.Handle(ctx, req)
+	if response.Allowed {
+		var patches []jsonpatch.Operation
+		for _, p := range response.Patches {
+			if p.Operation != "remove" {
```

Comment:
@alculquicondor would it be feasible to make the LossLessDefaulter more selective - only drop remove operations which correspond to fields outside of schema? If possible, then we could allow webhooks to remove known fields, while still preventing dropping of unknown fields. I suppose the answer may not be straightforward - depending how easily we can check if a remove operation corresponds to a known field, also performance may play a role. 

If possible, maybe it would increase chances for promoting LossLessDefaulter as part of opt-in defaulter inside controller-runtime?

Reply:
I suppose there is a way using the `reflect` package. But I don't think I'll have time to explore it.

Reply:
@mimowo That sounds more safety. Could we open an issue on the Kueue side?
If anyone is interested in this enhancement, they can take it.

Reply:
It would be wonderful, opened: https://github.com/kubernetes-sigs/kueue/issues/3174